daemon_test.go

```diff
@@ -68,112 +67,138 @@ func fakeProcessUpdate(response *http.Response) (interface{}, error) {
+// 	updater.fetchUpdateReturnError = errors.New("")
+// 	daemon := menderDaemon{}
+// 	daemon.Updater = updater
+
+// 	assert.False(t, fetchAndInstallUpdate(&daemon, UpdateResponse{}))
+// }
+
+// func Test_fetchAndInstallUpdate_installError_returnsNotInstalled(t *testing.T) {
+// 	updater := fakeUpdater{}
+// 	updater.GetScheduledUpdateReturnIface = new(UpdateResponse)
+// 	device := fakeDevice{}
+// 	device.retInstallUpdate = errors.New("")
+// 	daemon := menderDaemon{}
+// 	daemon.Updater = updater
+// 	daemon.UInstallCommitRebooter = device
+
+// 	assert.False(t, fetchAndInstallUpdate(&daemon, UpdateResponse{}))
+// }
+
+// func Test_fetchAndInstallUpdate_updatePartitionError_returnsNotInstalled(t *testing.T) {
+// 	updater := fakeUpdater{}
+// 	updater.GetScheduledUpdateReturnIface = new(UpdateResponse)
+// 	device := fakeDevice{}
+// 	device.retEnablePart = errors.New("")
+// 	daemon := menderDaemon{}
+// 	daemon.Updater = updater
+// 	daemon.UInstallCommitRebooter = device
+
+// 	assert.False(t, fetchAndInstallUpdate(&daemon, UpdateResponse{}))
+// }
+
+// func Test_fetchAndInstallUpdate_noErrors_returnsInstalled(t *testing.T) {
+// 	updater := fakeUpdater{}
+// 	updater.GetScheduledUpdateReturnIface = new(UpdateResponse)
+// 	device := fakeDevice{}
+// 	daemon := menderDaemon{}
+// 	daemon.Updater = updater
+// 	daemon.UInstallCommitRebooter = device
+
+// 	assert.True(t, fetchAndInstallUpdate(&daemon, UpdateResponse{}))
+// }
+
+// func Test_checkPeriodicDaemonUpdate_haveServerAndCorrectResponse_FetchesUpdate(t *testing.T) {
+
+// 	if testing.Short() {
+// 		t.Skip("skipping periodic update check in short tests")
+// 	}
+
+// 	reqHandlingCnt := 0
+// 	pollInterval := time.Duration(10) * time.Millisecond
+
+// 	// Test server that always responds with 200 code, and specific payload
+// 	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+// 		w.WriteHeader(204)
+// 		w.Header().Set("Content-Type", "application/json")
+// 		fmt.Fprintln(w, correctUpdateResponse)
+// 		reqHandlingCnt++
+// 	}))
+// 	defer ts.Close()
+
+// 	client, err := NewUpdateClient(httpsClientConfig{"client.crt", "client.key", "server.crt", true})
+// 	assert.NotNil(t, client)
+// 	assert.NoError(t, err)
+
+// 	device := NewDevice(nil, nil, "")
+// 	runner := newTestOSCalls("", 0)
+// 	controler := newTestMender(&runner)
+// 	if controler == nil {
+// 		t.Fatalf("failed to create mender")
+// 	}
+// 	controler.changeState(MenderStateBootstrapped)
+
+// 	daemon := NewDaemon(client, device, controler)
+// 	daemon.config = daemonConfig{serverpollInterval: pollInterval, serverURL: ts.URL}
+
+// 	go daemon.Run()
+
+// 	timespolled := 5
+// 	time.Sleep(time.Duration(timespolled) * pollInterval)
+// 	daemon.StopDaemon()
+
+// 	if reqHandlingCnt < (timespolled - 1) {
+// 		assert.Fail(t, "Expected to receive at least %v requests - %v received",
+// 			timespolled-1, reqHandlingCnt)
+// 	}
+// }
```

Comment:
Fixed. Removed most of those, reworked the last one to check if actual mender state sequence was run.